controllers/aggregated_clusteroperator_controller.go

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	openshiftconfigv1 "github.com/openshift/api/config/v1"
+	configv1client "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	utilerror "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/client-go/discovery"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	logr "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	platformv1alpha1 "github.com/openshift/platform-operators/api/v1alpha1"
+	aggregatedco "github.com/openshift/platform-operators/internal/aggregated-co"
+	"github.com/openshift/platform-operators/internal/util"
+)
+
+type AggregatedClusterOperatorReconciler struct {
+	client.Client
+	DiscoveryClient discovery.DiscoveryInterface
+	Configv1Client  configv1client.ConfigV1Interface
+}
+
+const aggregateCOName = "platform-operators-aggregated"
+
+//+kubebuilder:rbac:groups=platform.openshift.io,resources=platformoperators,verbs=get;list;watch
+//+kubebuilder:rbac:groups=platform.openshift.io,resources=platformoperators/status,verbs=get
+//+kubebuilder:rbac:groups=config.openshift.io,resources=clusteroperators,verbs=get;list;watch;create;update;patch;delete
```

Comment:
nit: Do we need access to all these verbs, or can we reduce them to the subset we need for this controller?